### PR TITLE
Safari items!

### PR DESCRIPTION
**Pull request by Jaaslet**
A small step towards making safari more fun!
Items can randomly spawn on the ground, like pokemon currently do in the safari. And you can pick them up, by walking over them!
What items you ask? Not my job to figure out! Just adding the code. Then someone else can fill out the list. But i think there will be at least one mega stone!

### CODE REVIEW

**Comment by Jaaslet** [comment]
PR is ready for review now.
Each time you leave a battle, there is a 50/50 chance that an item will spawn from a weighted array. I am considering removing so run can spawn an item, but we can talk about that, when friend safari is ready for crew test.
I have added two items to friend safari (which is not out yet). Hopefully someone else wants to figure out which items we want to put in the two safaris

**Comment by Farboo171** [comment]
> Hopefully someone else wants to figure out which items we want to put in the two safaris

it me

File: src/scripts/items/BagHandler.ts [hunk 1]
```diff
@@ -90,6 +90,22 @@ class BagHandler {
         }
     }
 
+    public static isAvailable(item: BagItem): boolean {
+        if (!item) {
+            return false;
+        }
+        switch (item.type) {
```
Comment (Aegyo) [inline src/scripts/items/BagHandler.ts:97]: switch with no default makes me nervous

**Review by Aegyo — COMMENTED** [review]
The image desperately needs some alpha
Screenshot_2023-05-07_22-33-08

**Comment by DraKay** [comment]
The image soul posted on discord have transparent background

**Comment by Jaaslet** [comment]
> The image soul posted on discord have transparent background

I know. I just forgot to upload it

**Comment by Farboo171** [comment]
Safari Zone Items:
(Canon): Leaf stone, Large Restore, Medium Restore, Protein, Max Revive, Revive
(Clicker): Fire stone, Thunder stone, Water stone, Pinsirite, Rare Candy, some gen 1 and 2 berries as filler items

Friend Safari Items:
No canon items
(Clicker): Scizorite, Rare Candy, Carbos, Calcium, Star Pieces, Shiny Stones, Dusk Stones, Repeat Balls, Dusk Balls, Luxury Balls, gen 2 and 3 berries as filler
(posted here so i don't lose the list)

**Comment by Jaaslet** [comment]
Fixed the image, made the switch less scary and also found out in testing, that items and pokemon could spawn on the same area. So fixed that too

**Comment by Jaaslet** [comment]
Turned it into a draft, as i want to add requirement and do a little refactoring. There is no reason for us to select what item it is, when it's spawns. It can lead to problems with duplicate mega stones. Might as well select it when the player picks it up

**Comment by Jaaslet** [comment]
Stuff fixed. Ready for review again

**Comment by Aegyo** [comment]
Now it's looking a little blurry
Screenshot_2023-05-11_02-44-36

Seems fine other than that though so I'll approve and you can either deal with the image now or another time.

**Review by Aegyo — APPROVED** [review]

**Review by DraKay — COMMENTED** [review]
It is trying to generate a 32x32 tile from a 16x16 image. Try with this version (just copy original with 2x2 pixels that's the same line style as pokes have)

export1683788423173

**Pull request merged**